File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int, int> values;
        int ans = 0;

        for(auto it: nums)
        {
            values[it] = 1;
        }

        for(auto it: nums)
        {
            if(values.find(it-1) != values.end())
            {
                continue;
            }
            else
            {
                int temp = it;
                int count = 0;
                while(values.find(temp) != values.end())
                {
                    count++;
                    temp++;
                }
                ans = max(ans, count );
            }
        }

        return ans;

    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] arr) {
        int count = 1;
        int element = 0;
        int n = arr.length;
        if(n == 0)return 0;
        int res = Integer.MIN_VALUE;
        Arrays.sort(arr);
        for(int i=1;i<n;i++){
            if(arr[i-1] == arr[i]){
                continue;
            }else if(arr[i-1]+1 == arr[i]){
                count++;
            }else{
                res = Math.max(res, count);
                count = 1;
            }
        }
        res = Math.max(res, count);
        return res;
    }
}